vue/components/form/inputs/conditional-visibility.ts

```diff
@@ -0,0 +1,46 @@
+import Operator from '../definitions/field-operator';
+
+import FormPayload = APIConsumer.FormPayload;
+
+export type Statement = [ string | Statement, Operator, any | Statement ] | null;
+
+function calculateVisibility(payload: FormPayload, statement: Statement): boolean {
+  if (statement == null) return true;
+
+  const operator = statement[1];
+
+  if (Operator.isLogicOperator(operator)) {
+    // we have to go 1 level lower
+    switch (statement[1]) {
+    case Operator.AND:
+      return calculateVisibility(payload, statement[0] as Statement) && calculateVisibility(payload, statement[2]);
+    case Operator.OR:
+      return calculateVisibility(payload, statement[0] as Statement) || calculateVisibility(payload, statement[2]);
+    default:
+      throw new Error(`Not implemented operator ${operator}`);
```

Comment:
??? zakaj pa to? saj tisti operatorji so vsi logical. Njihova implementacija je trivialna....